src/codegen/llvm/codegen_llvm.cc

```diff
@@ -113,7 +113,7 @@ void CodeGenLLVM::AddFunctionInternal(const LoweredFunc& f, bool ret_void) {
       ret_void ? t_void_ : t_int_, arg_type, false);
   // setup the function.
   function_ = llvm::cast<llvm::Function>(module_->getOrInsertFunction(f->name, ftype));
-  function_->setCallingConv(llvm::CallingConv::C);
+  function_->setCallingConv(dev_type == AMDGPU ? llvm::CallingConv::AMDGPU_KERNEL : llvm::CallingConv::C);
```

Comment:
Likely you don't have to do this, we can overwrite AddFunction, and do setCallingConv after calling AddFunctionInternal

Reply:
I don't want to create multiple functions which do the same and it is easier to read the code this way. Do you want me to overload the function?

Reply:
We don;t have to overload AddFunctionInternal. Simply overload AddFunction, which calls AddFunctionInternal, and then do function_->setCallingConv again